cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -257,7 +262,8 @@ func (uc *UpstreamController) updateRuleStatus() {
 				errSlice := make([]string, 0)
 				rule.Status.Errors = append(errSlice, content.Error.Detail)
 			}
-			newStatus := &rulesv1.RuleStatus{}
+			newStatus := &RuleStatus{}
+			newStatus.Status = rule.Status
 			body, err := json.Marshal(newStatus)
```

Comment:
Thanks for the fixing, and has any test here? Is the status worked as desired? For the patch func, the data should be patch bytes created through olddata and newdata. 

Ref: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go#L346